### ISSUE TITLE
Unicode characters in .dx files supported on Mac, not on Linux

### ISSUE THREAD

**Issue by harryaskham** [google-research/dex-lang#559]
Discovered trying to move a file with the word "Schrödinger" in a comment from MacOS to Debian :)

On MacOS everything works as expected, umlaut appears in Web view

On Linux
```
Error:
examples/unicode.dx: hGetContents: invalid argument (invalid byte sequence)
```

Removing the umlaut fixes things on Linux.

**Comment by apaszke** [comment]
Yeah we're very unprincipled when it comes to unicode handling. We should take a pass over the whole compiler and be more careful about it.
